service/src/java/org/apache/hive/service/auth/PlainSaslHelper.java

```diff
@@ -65,6 +70,15 @@ public static TTransportFactory getPlainTransportFactory(String authTypeStr)
     return saslFactory;
   }
 
+  public static TTransportFactory getDualPlainTransportFactory(TTransportFactory otherTrans,
+                                                               String hiveHost)
+          throws LoginException {
+    LOG.info("Created additional transport factory for skipping authentication when client " +
```

Comment:
Maybe we should perform a validation of the input parameters (not null, not empty host string) here?

Reply:
As per the latest approach, the trusted domain will be configured and thus we do not rely on hivehost to know the trusted domain. So hiveHost won't be passed.